databricks/koalas/frame.py

```diff
@@ -7247,6 +7248,40 @@ def append(
 
         return cast(DataFrame, concat([self, other], ignore_index=ignore_index))
 
+    def combine_first(self, other: "DataFrame") -> "DataFrame":
+        if isinstance(other, ks.Series):
+            other = other.to_frame()
+
+        update_columns = list(
+            set(self._internal.column_labels).intersection(set(other._internal.column_labels))
+        )
+
+        update_sdf = combine_frames(self, other)._internal.resolved_copy.spark_frame
+
+        for column_labels in update_columns:
+            column_name = self._internal.spark_column_name_for(column_labels)
+            old_col = scol_for(update_sdf, "__this_" + column_name)
+            new_col = scol_for(
+                update_sdf, ("__that_" + other._internal.spark_column_name_for(column_labels))
+            )
+            update_sdf = update_sdf.withColumn(
+                "__this_" + column_name, F.when(old_col.isNull(), new_col).otherwise(old_col)
+            )
+            update_sdf = update_sdf.drop("__that_" + column_labels[0])
```

Comment:
Maybe we can simply select the necessary columns rather than use `withColumn` and `drop` ??

like

```python
spark_columns = combined_df._internal.spark_columns
# Add some code to exclude `column_labels[0]` here
spark_columns = ...
cond = F.when(old_col.isNull(), new_col).otherwise(old_col).alias("__this_" + column_name)

update_sdf.select(*spark_columns, cond)
```